core/src/main/java/org/mule/runtime/core/api/processor/InterceptingMessageProcessor.java

```diff
@@ -15,14 +17,26 @@
  * </p>
  * Although not normal, it is valid for the <i>listener</i> MessageProcessor to be <i>null</i> and implementations should handle
  * this case.
- * 
+ *
  * @since 3.0
+ *
+ * @deprecated Use interception API instead of this interface. Ref: {@link ProcessorInterceptorFactory}.
  */
+@Deprecated
 public interface InterceptingMessageProcessor extends Processor {
 
   /**
    * Set the MessageProcessor listener that will be invoked when a message is received or generated.
    */
   void setListener(Processor listener);
 
+  /**
+   * Indicates if the processing strategy of the parent flow is applied to the intercepted processor's chain.
+   *
+   * @return {@code true} if the {@code listener} will run wholly in the same thread and blocking if needed. {@code false} if the
+   *         lfow's {@link ProcessingStrategy} will be used.
+   */
+  default boolean isBlocking() {
```

Comment:
Consider:
- Using BlockingProcessorStrategy in child chain for CXF instead of this flag.
- If we think this may be a common scenario think about if it makes sense to represent this as a processing type.